src/basic-samples/GuaranteedSubscriber.js

```diff
@@ -0,0 +1,254 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Solace Web Messaging API for JavaScript
+ * Receiving Guaranteed messages on a Topic tutorial - Guaranteed Subscriber
+ * Demonstrates receiving persistent messages on a topic
+ */
+
+/*jslint es6 browser devel:true*/
+/*global solace*/
+
+var GuaranteedSubscriber = function (solaceModule, queueName, topicName ) {
```

Comment:
hmmm looks like `topicName` is not used at all - am I wrong? The GuaranteedSubscriber should just bind to a queue (and optionally add topic subscription to the queue)

Reply:
Correct - topicName is not required (or used); removed the references. The assumption is the queue with a subscription is pre-created before running the GuaranteedSubscriber sample. This is in line with QueueConsumer (just as a reference).

Do you think we should add code to create a subscription - IMO that would be covered in the tutorial. Wdyt?

Reply:
This is the same case with JavaScript samples - let me know whether adding a subscription at the code level makes sense or if we should leave it in the description of the tutorial. IMO, we are asking them to create a queue anyways, just extending that to add a subscription would be fine.

Reply:
Let's leave it as is then, and I'll create a new TopicToQueueMapping sample to showcase how to add subs to queues

Reply:
Okay.